libpromises/evalfunction.c

```diff
@@ -2931,7 +2931,10 @@ JsonElement *DefaultTemplateData(const EvalContext *ctx, const char *wantbundle)
             if (NULL != scope_obj)
             {
                 char *lval_key = VarRefToString(var->ref, false);
-                JsonObjectAppendElement(scope_obj, lval_key, RvalToJson(var->rval));
+                if (NULL == strchr(lval_key, '#')) // don't collect mangled refs
+                {
+                    JsonObjectAppendElement(scope_obj, lval_key, RvalToJson(var->rval));
+                }
```

Comment:
I vaguely recall that there are some exceptions to this rule, and '#' is used to mark those exceptions, but I don't remember any details (or even if I remember correctly..). I suppose we're ok if the tests pass, and this is anyway limited to the template data.